pkg/rancher-desktop/backend/k8s.ts

```diff
@@ -123,7 +123,7 @@ export interface KubernetesBackend extends EventEmitter<KubernetesBackendEvents>
    * Calculate any reasons that may require us to restart the backend, had the
    * given new configuration been applied on top of the existing old configuration.
    */
-  requiresRestartReasons(oldConfig: BackendSettings, newConfig: RecursivePartial<BackendSettings>, extras?: ExtraRequiresReasons): Promise<RestartReasons>;
+  requiresRestartReasons(oldConfig: BackendSettings, newConfig: RecursiveReadonly<BackendSettings>, extras?: ExtraRequiresReasons): Promise<RestartReasons>;
```

Comment:
I think I backed out because of a typescript message in lima.ts at the call:
```
TS2345: Argument of type 'RecursivePartial<RecursiveReadonly<{ 
   version: number;
   application: {
      adminAccess: boolean;
      debug: boolean;
      pathManagementStrategy: PathManagementStrategy;
      telemetry: { enabled: boolean; };
      updater: { ...; };
    };
    ... 10 more ...;
    window: { ...; };
  }>>' is not assignable to parameter of type 

'RecursiveReadonly<RecursiveReadonly<{
  version: number;
  application: {
    adminAccess: boolean;
    debug: boolean;
    pathManagementStrategy: PathManagementStrategy;
    telemetry: { enabled: boolean; };
    updater: { ...; };
  };
  ... 10 more ...;
  window: { ...; };
}>>'


Types of property 'version' are incompatible.

Type 'number | undefined' is not assignable to type 'number'.
Type 'undefined' is not assignable to type 'number'.
```

`BackendSettings` == `RecursiveReadonly<Settings>` so `RecursivePartial` is needed to allow for missing fields.